listenbrainz/webserver/static/js/src/UserArtist.tsx

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable */
+import * as ReactDOM from "react-dom";
+import * as React from "react";
+
+import Bar from "./Bar";
+import APIService from "./APIService";
+
+// const data = [
+//   {
+//     id: "The Chainsmokers & Coldplay & Alan Walker",
+//     Listens: 590,
+//   },
+//   {
+//     id: "Ellie Goulding",
+//     Listens: 350,
+//   },
+//   {
+//     id: "The Fray",
+//     Listens: 320,
+//   },
+//   {
+//     id: "Vance Joy",
+//     Listens: 100,
+//   },
+//   {
+//     id: "Lenka",
+//     Listens: 50,
+//   },
+//   {
+//     id: "Ritviz",
+//     Listens: 45,
+//   },
+// ];
+
+export type UserArtistProps = {
+  user: ListenBrainzUser;
+  apiUrl: string;
+};
+
+export type UserArtistState = {
+  data: Array<{
+    id: string;
+    Listens: number;
+  }>;
+};
+
+export default class UserArtist extends React.Component<
+  UserArtistProps,
+  UserArtistState
+> {
+  APIService: APIService;
+
+  private ROWS_PER_PAGE = 25;
+
+  constructor(props: UserArtistProps) {
+    super(props);
+
+    this.APIService = new APIService(
+      props.apiUrl || `${window.location.origin}/1`
+    ); // Used to access LB API
+
+    this.state = {
+      data: [] as UserArtistState["data"],
+    };
+  }
+
+  processData = (data: any): UserArtistState["data"] => {
```

Comment:
Let's define `UserArtistState["data"]` as an individual type instead of doing this. It's hard to understand.